File: 05_speed2.f90
program main
    use iso_c_binding
    implicit none
    integer, parameter :: n = 10000, m = 10000
    real(c_double), dimension(n, m) :: array1, array2, result
    integer :: i, j
    real(c_double) :: start, finish, elapsed

    ! Initialize 2D arrays
    do i = 1, n
        do j = 1, m
            array1(i, j) = real(i + j, c_double)
            array2(i, j) = real(n - i + m - j, c_double)
        end do
    end do

    call cpu_time(start)
    ! Perform element-wise addition
    result = array1 / 100.D0  + array2**2.D0 + sin(array1 / 100.D0)
    call cpu_time(finish)

    elapsed = finish - start
    print *, "Operation took ", elapsed, " seconds."

    ! Optionally print some results
    print *, "Sample result: ", result(n, m)
end program main

File: 05_speed2.cpp
#include <iostream>
#include <vector>
#include <chrono>

using namespace std;

int main() {
    const int n = 10000, m = 10000;
    vector<vector<double>> array1(n, vector<double>(m)),
                                     array2(n, vector<double>(m)),
                                     result(n, vector<double>(m));

    // Initialize 2D arrays
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < m; ++j) {
            array1[i][j] = i + j + 1;
            array2[i][j] = n - i + m - j - 1;
        }
    }

    auto start = chrono::high_resolution_clock::now();
    // Perform element-wise addition
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < m; ++j) {
            result[i][j] = array1[i][j] + array2[i][j];
        }
    }
    auto finish = chrono::high_resolution_clock::now();

    chrono::duration<double> elapsed = finish - start;
    cout << "Operation took " << elapsed.count() << " seconds.\n";

    // Optionally print some results
    cout << "Sample result: " << result[n/2][m/2] << endl;

    return 0;
}
